### ISSUE TITLE
`issquare` does not work with non-monic defining polynomials

### ISSUE THREAD

**Issue by thofma** [thofma/Hecke.jl#229]
```julia
julia> Qx,x = Hecke.QQ["x"]
(Univariate Polynomial Ring in x over Rational Field, x)
julia> f = 8x^3 + 4x^2 - 4x - 1
8*x^3 + 4*x^2 - 4*x - 1
julia> isirreducible(f)
true
julia> K,a = NumberField(f,"a")
(Number field over Rational Field with defining polynomial 8*x^3 + 4*x^2 - 4*x - 1, a)
julia> issquare(K(1))
ERROR: Primitive element must be integral
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] __equation_order(::AnticNumberField) at /home/r/.julia/packages/Hecke/tHGcU/src/NfOrd/NfOrd.jl:1001
 [3] EquationOrder(::AnticNumberField, ::Bool) at /home/r/.julia/packages/Hecke/tHGcU/src/NfOrd/NfOrd.jl:976
 [4] EquationOrder at /home/r/.julia/packages/Hecke/tHGcU/src/NfOrd/NfOrd.jl:970 [inlined]
 [5] _roots_hensel(::AbstractAlgebra.Generic.Poly{nf_elem}; max_roots::Int64, ispure::Bool, isnormal::Bool, root_bound::Array{fmpz,1}) at /home/r/.julia/packages/Hecke/tHGcU/src/NfOrd/Hensel.jl:236
 [6] roots(::AbstractAlgebra.Generic.Poly{nf_elem}; max_roots::Int64, ispure::Bool, isnormal::Bool) at /home/r/.julia/packages/Hecke/tHGcU/src/NumField/NfAbs/Elem.jl:702
 [7] ispower(::nf_elem, ::Int64; with_roots_unity::Bool, isintegral::Bool, trager::Bool) at /home/r/.julia/packages/Hecke/tHGcU/src/NumField/NfAbs/Elem.jl:777
 [8] ispower at /home/r/.julia/packages/Hecke/tHGcU/src/NumField/NfAbs/Elem.jl:751 [inlined]
 [9] issquare(::nf_elem) at /home/r/.julia/packages/Hecke/tHGcU/src/NumField/NfAbs/Elem.jl:841
```

**Comment by thofma** [comment]
Random test shows that this is broken
```julia
julia> f = -1//4*x^2 + 8//9*x + 6//7
-1//4*x^2 + 8//9*x + 6//7

julia> K, a = number_field(f)
(Number field over Rational Field with defining polynomial -1//4*x^2 + 8//9*x + 6//7, _a)

julia> K(175//72*x + 775//112)
175//72*_a + 775//112

julia> b = K(175//72*x + 775//112)
175//72*_a + 775//112

julia> issquare(b)
Flint exception (Impossible inverse):
    Cannot invert modulo 7*1
ERROR: Problem in the Flint-Subsystem
Stacktrace:
  [1] error(s::String)
    @ Base ./error.jl:33
  [2] flint_abort()
    @ Nemo ~/.julia/packages/Nemo/xNg6C/src/Nemo.jl:118
  [3] fmpq_poly_to_gfp_poly_raw!
    @ ~/.julia/dev/Hecke/src/Misc/Poly.jl:447 [inlined]
  [4] fmpq_poly_to_gfp_poly(Rx::GFPPolyRing, f::fmpq_poly)
    @ Hecke ~/.julia/dev/Hecke/src/Misc/Poly.jl:480
  [5] GFPPolyRing
    @ ~/.julia/dev/Hecke/src/Misc/ResidueRingPoly.jl:199 [inlined]
  [6] _roots_hensel(f::AbstractAlgebra.Generic.Poly{nf_elem}; max_roots::Int64, ispure::Bool, isnormal::Bool, root_bound::Vector{fmpz})
    @ Hecke ~/.julia/dev/Hecke/src/NfOrd/Hensel.jl:172
  [7] roots(f::AbstractAlgebra.Generic.Poly{nf_elem}; max_roots::Int64, ispure::Bool, isnormal::Bool)
    @ Hecke ~/.julia/dev/Hecke/src/NumField/NfAbs/Elem.jl:702
  [8] ispower(a::nf_elem, n::Int64; with_roots_unity::Bool, isintegral::Bool, trager::Bool)
    @ Hecke ~/.julia/dev/Hecke/src/NumField/NfAbs/Elem.jl:779
  [9] ispower
    @ ~/.julia/dev/Hecke/src/NumField/NfAbs/Elem.jl:751 [inlined]
```
Or:
```julia
julia> f = 7//4*x^2 - 1//4*x - 7//6
K, a = 7//4*x^2 - 1//4*x - 7//6

julia> K, a = NumberField(f)
(Number field over Rational Field with defining polynomial 7//4*x^2 - 1//4*x - 7//6, _a)

julia> c = K(-24//343*x + 11//147)
-24//343*_a + 11//147

julia> issquare(c)
(parent((parent(a)).pol))(a) = -24//343*x + 11//147
parent(a) = Number field over Rational Field with defining polynomial 7//4*x^2 - 1//4*x - 7//6
ERROR: Denominator has to be 1
Stacktrace:
 [1] error(s::String)
   @ Base ./error.jl:33
 [2] (::FmpzPolyRing)(b::fmpq_poly)
   @ Hecke ~/.julia/dev/Hecke/src/Misc/Poly.jl:611
 [3] _hensel(f::AbstractAlgebra.Generic.Poly{nf_elem}, fac_pol_mod_p::gfp_poly, fp::fq_nmod_poly, k::Int64; max_roots::Int64, ispure::Boo
l, isnormal::Bool)
   @ Hecke ~/.julia/dev/Hecke/src/NfOrd/Hensel.jl:426
 [4] _roots_hensel(f::AbstractAlgebra.Generic.Poly{nf_elem}; max_roots::Int64, ispure::Bool, isnormal::Bool, root_bound::Vector{fmpz})
   @ Hecke ~/.julia/dev/Hecke/src/NfOrd/Hensel.jl:271
 [5] roots(f::AbstractAlgebra.Generic.Poly{nf_elem}; max_roots::Int64, ispure::Bool, isnormal::Bool)
   @ Hecke ~/.julia/dev/Hecke/src/NumField/NfAbs/Elem.jl:702
 [6] ispower(a::nf_elem, n::Int64; with_roots_unity::Bool, isintegral::Bool, trager::Bool)
   @ Hecke ~/.julia/dev/Hecke/src/NumField/NfAbs/Elem.jl:779
 [7] ispower
   @ ~/.julia/dev/Hecke/src/NumField/NfAbs/Elem.jl:751 [inlined]
 [8] issquare(a::nf_elem)
   @ Hecke ~/.julia/dev/Hecke/src/NumField/NfAbs/Elem.jl:841
 [9] top-level scope
   @ REPL[11]:1
```

**Comment by thofma** [comment]
Closed with #239.
